core/variables.js

```diff
@@ -512,18 +505,17 @@ exports.getVariable = getVariable;
  *     or name + type combination.
  */
 const createVariable = function(workspace, id, opt_name, opt_type) {
-  const potentialVariableMap = workspace.getPotentialVariableMap();
   // Variables without names get uniquely named for this workspace.
   if (!opt_name) {
     const ws = workspace.isFlyout ? workspace.targetWorkspace : workspace;
     opt_name = exports.generateUniqueName(ws);
   }
 
-  // Create a potential variable if in the flyout.
+  // Create a variable directly in the variable map if in the flyout.
```

Comment:
This section is where the refactor would simplify things. `workspace.getVariableMap().createVariable` is equivalent to `workspace.createVariable`. No matter if we are in a flyout or a main workspace, after this refactor, we would just create it in the variable map so we could eliminate the if statements here and just do

```js
variable = workspace.createVariable(...)
```

However again, looking at this simplification vs the additional complexity introduced when looking for a variable, I'm not sure it's worth doing.

Reply:
I concur about the change required: the entire second section of the function can be replace by:
```JS
  return workspace.createVariable(opt_name, opt_type, id);
```
(which is in part why I think this change _is_ worth doing).